src/audio/kpb.c

```diff
@@ -784,12 +797,21 @@ static int kpb_params(struct comp_dev *dev,
 static int kpb_prepare(struct comp_dev *dev)
 {
 	struct comp_data *kpb = comp_get_drvdata(dev);
+	struct sof_ipc_stream_params params;
 	int ret = 0;
 	int i;
 	size_t hb_size_req = KPB_MAX_BUFFER_SIZE(kpb->config.sampling_width, kpb->config.channels);
 
 	comp_dbg(dev, "kpb_prepare()");
 
+	kpb_set_params(dev, &params);
+
+	ret = kbp_verify_params(dev, &params);
```

Comment:
the naming of both of these functions isn't very... intuitive. AFAICS `kpb_set_params()` actually doesn't **set** parameters but rather **gets** them instead. I.e. it retrieves current KPB parameters into a user-provided structure. Then it isn't very clear why `kbp_verify_params()` is called - if you've just obtained parameters from a device, they should be valid, right? But also instead of just verifying parameters, which sounds like a read-only operation, `kbp_verify_params()` and the underlying `comp_verify_params()` can update parameters and also use them to configure audio buffers... Maybe we need at least some comments here and eventually also in the implementation.